public/javascripts/forum-post.js

```diff
@@ -9,4 +9,68 @@ $(function () {
     form[0].reset();
     form.find('textarea').height(message.height());
   });
+
+  lichess.loadScript('/assets/javascripts/vendor/jquery.textcomplete.js').then(function() {
+
+    var getTopicId = function () {
+        return $('.post-text-area').attr('data-topic');
+    };
+
+    var getThreadParticipants = function() {
+        var topicId = getTopicId();
+
+        if (!topicId) {
+            return jQuery.Deferred().resolve([]);
+        } else {
+            return $.ajax({url: "/forum/participants/" + topicId});
+        }
+
+    };
+
+    var searchCandidates = function(term, candidateUsers) {
+        return $.map(candidateUsers,
```

Comment:
I tried it out and it looks like the reason the jquery-textcomplete examples use jQuery's map ($.map) is because returning null from the function applied to an element removes the item from the map.

Using candidateUsers.map(function() ...) results in 'null' being displayed.